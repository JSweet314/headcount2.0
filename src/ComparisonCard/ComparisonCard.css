.comparison-card {
  height: 450px;
  width: 320px;
  vertical-align: bottom;
  border: 6px solid #309B96;
}

.comparison-card header {
  padding: 10px 0;
  width: 100%;
  height: 100px;
  border-top-right-radius: 19px;
  border-top-left-radius: 19px;
  background-color: #309B96;
  font-size: 1.3em;
  text-align: center;
  vertical-align: top;
}

.comparison-card main{
  height: 100%;
}

.comparison-card footer {
  padding: 10px 0;
  border-bottom-right-radius: 19px;
  border-bottom-left-radius: 19px;
  background-color: #309B96;
 }

 .comparedAverage__text {
  text-align: center;
  padding:  60px 20px;
 }

 .comparedAverage__num {
  position: relative;
  bottom: 40px; 
  margin: 0;
  font-size: 65px;
  font-weight: bold;
  text-align: center;
  color:#fc7753bd;
 }

.percent {
  font-size: 25px;
  text-align: center;
  margin: 0;
}

.comparison__h2 {
  width: 100%;
  margin: 0;
  font-size: 30px;
  text-align: center;
}
 
@media screen and (max-width: 1300px) {
  .comparison-card {
    margin: 3px auto;
  }
}

@media screen and (max-width: 800px) {
  .comparison-card {
    margin: 10px auto;
  }
}

