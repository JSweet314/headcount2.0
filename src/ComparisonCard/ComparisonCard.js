import React from 'react';
import PropTypes from 'prop-types';

const ComparisonCard = ({comparedAverage}) => {
  const [locationNameA, locationNameB, compared] = Object.keys(comparedAverage);

  return (
    <article>
      <h2>{locationNameA}</h2>
      <h4>{comparedAverage[locationNameA]}</h4>
      <h3>Compared Averages: {comparedAverage[compared]}</h3>
      <h2>{locationNameB}</h2>
      <h4>{comparedAverage[locationNameB]}</h4>
    </article>
  );
};	

ComparisonCard.propTypes = {
  comparedAverage: PropTypes.object.isRequired
};


export default ComparisonCard;