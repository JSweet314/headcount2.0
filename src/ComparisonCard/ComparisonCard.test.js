import React from 'react';
import {shallow} from 'enzyme';
import ComparisonCard from './ComparisonCard.js';


describe('ComparisonCard', () => {
  let wrapper;
  const comparedAverage = { 
    COLORADO:0.34, 
    'COLORADO SPRINGS': 0.45, 
    compared:0.75 
  };

  beforeEach(() => {
    wrapper = shallow(
      <ComparisonCard comparedAverage={comparedAverage}/>);
  });

  it('Should match the snapshot', ()=> {
    expect(wrapper).toMatchSnapshot();
  });
});