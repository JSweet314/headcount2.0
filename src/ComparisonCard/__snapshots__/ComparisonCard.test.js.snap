// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonCard Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    comparedAverage={
        Object {
            "COLORADO": 0.34,
            "COLORADO SPRINGS": 0.45,
            "compared": 0.75,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          COLORADO
</h2>,
        <h4>
          0.34
</h4>,
        <h3>
          Compared Averages: 
          0.75
</h3>,
        <h2>
          COLORADO SPRINGS
</h2>,
        <h4>
          0.45
</h4>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.34,
        },
        "ref": null,
        "rendered": 0.34,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Compared Averages: ",
            0.75,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Compared Averages: ",
          0.75,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO SPRINGS",
        },
        "ref": null,
        "rendered": "COLORADO SPRINGS",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.45,
        },
        "ref": null,
        "rendered": 0.45,
        "type": "h4",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            COLORADO
</h2>,
          <h4>
            0.34
</h4>,
          <h3>
            Compared Averages: 
            0.75
</h3>,
          <h2>
            COLORADO SPRINGS
</h2>,
          <h4>
            0.45
</h4>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.34,
          },
          "ref": null,
          "rendered": 0.34,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Compared Averages: ",
              0.75,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Compared Averages: ",
            0.75,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO SPRINGS",
          },
          "ref": null,
          "rendered": "COLORADO SPRINGS",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.45,
          },
          "ref": null,
          "rendered": 0.45,
          "type": "h4",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
